src/bin/ion/commands/beta/generate/templates/java/anonymous_type.templ

```diff
@@ -0,0 +1,209 @@
+{# following macro defines an anonymous type as children class for its parent type definition #}
```

Comment:
(🗺️ PR tour) This file adds 3 new macros for generating anonymous types.
* `anonymous_type`: Generates a new child class for given anonymous type based on its properties (e.g. `target_kind_name`, `abstract_data_type`). 
* `initialize_anonymous_type`: This is helper macro to initialize anonymous types inside setter methods.
* `define_params_for_anonymous_type`: This is helper macro to define arguments to setter methods. (This macrohelps get the base type as argument instead of anonymous type itself, check [this code snippet](https://github.com/amazon-ion/ion-cli/pull/109/files#diff-8fddb5eb93f33a051ff30f87fc97456d02b921ed0d46cb4b78bd889d7c33cb75L42-R48) to see how this is used)

Reply:
Can we just get rid of the anonymous types when it's a `List<Something>`? (Is that going to be done in a different PR?)

Reply:
Did you mean to get rid of creating a nested class which contains `List<Something>`? I have created a separate class for anonymous type with ISL like `{type: list, element: int}` because there could be more constraints added to this anonymous type which might require adding more variables/APIs to this anonymous type class with `List<Something>`.
But I can add a check to see if there are no more constraints then the `type` and `element` then I can skip generating anonymous type for it.